theseus/tests/test_theseus_layer.py

```diff
@@ -280,11 +301,44 @@ def cost_weight_fn():
             | (info.status == th.NonlinearOptimizerStatus.FAIL)
         ).all()
 
-        loss = F.mse_loss(pred_vars["coefficients"], target_vars["coefficients"])
+        if learning_method == "leo":
+            # groundtruth cost
+            x_gt = target_vars["coefficients"]
+            input_values_gt = {
+                "coefficients": x_gt,
+                cost_weight_param_name: cost_weight_fn(),
+            }
+            layer_to_learn.objective.update(input_values_gt)
+            cost_gt = torch.sum(layer_to_learn.objective.error(), dim=1)  # B x 1
+            cost_gt = cost_gt.to(device)
```

Comment:
Might not be the best place to move between device in the learning loop. This line won't be needed if inputs, objective, layer etc are setup to be on the correct device at the beginning.

Reply:
Agreed, removed the to(device)